### ISSUE TITLE
supabase-studio is unhealthy when self-hosting on Ubuntu

### ISSUE THREAD

**Issue by outSH** [supabase/supabase#22867]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 
  - Similar issue here: https://github.com/orgs/supabase/discussions/21558

## Describe the bug

I can't run supabase in docker using any Ubuntu machine (checked multiple environments) because studio is unhealthy. Studio container doesn't contain any helpful information (just that it's listening on port 3000), I can provide debug logs if you let me know how to enable them :)

## To Reproduce

1. Follow https://supabase.com/docs/guides/self-hosting/docker on any environment listed in system information below.
2. The supabase-studio is unhealthy

## Expected behavior

All containers should be healthy.

## System information

- OS:
  - Ubuntu 22.04.3 LTS on WSL2
  - Ubuntu 20.04.6 LTS VM on HyperV
  - Docker in docker (docker:25.0.5-dind) on both machines above.
- Branches: `master` / `v0.24.03`. Occurred on older tags as well but I did not keep track of them.

## Additional context

It used to work in v0.22.10, but this version can't be used anymore because it's docker compose file references image that is not available anymore. We use supabase in DIND container for testing purposes in Hyperledger Cacti project - https://github.com/hyperledger/cacti/tree/main/tools/docker/supabase-all-in-one.

**Comment by outSH** [comment]
@garrettcam Thanks, it does solve the issue!

However I'd prefer using the docker-compose in the repo without a change. I think the root cause might be caused by node 17+ binding to ipv6 by default:

```
  ▲ Next.js 13.5.6
  - Local:        http://54207504f03d:3000 # should be 127.0.0.1 for localhost ping to work
  - Network:      http://172.25.0.8:3000

 ✓ Ready in 293ms
```

**Comment by Hallidayo** [comment]
Hi all, We’re going to close this one due to inactivity. Please feel free to reopen if you’re still having issues here or open a new issue.

**Comment by Jack74r** [comment]
Hi, idk if i can reply here or reopen issue but issue looks like this one. (maybe miss conf on my side)

I have edit the .env to change the kong port because the 8000 is already in use. Everything looks working but i have container storage and studio unhealty.

My local serveur run on 192.168.1.155

My change in .env :

```
KONG_HTTP_PORT=8442
DISABLE_SIGNUP=true
SUPABASE_PUBLIC_URL=http://192.168.1.155:8442
```


My log of studio container log
  ▲ Next.js 14.2.3
  
Local:        http://7bf4e0dc9316:3000/
Network:      http://172.30.0.11:3000/
✓ Starting...
✓ Ready in 540ms


inscpect supabase storage : 
```
  "Health": {
                "Status": "unhealthy",
                "FailingStreak": 225,
                "Log": [
                    {
                        "Start": "2024-09-09T15:15:55.768547175+02:00",
                        "End": "2024-09-09T15:15:55.831509442+02:00",
                        "ExitCode": 1,
                        "Output": "Connecting to localhost:5000 ([::1]:5000)\nwget: can't connect to remote host: Connection refused\n"
```
i think the problem is local ip is wrong : Local:        http://7bf4e0dc9316:3000/ but i dont know why

Studio inspect
```
"Health": {
                "Status": "unhealthy",
                "FailingStreak": 505,
                "Log": [
                    {
                        "Start": "2024-09-09T15:39:50.901089062+02:00",
                        "End": "2024-09-09T15:39:50.992584645+02:00",
                        "ExitCode": 1,
                        "Output": "node:events:497\n      throw er; // Unhandled 'error' event\n      ^\n\nError: connect ECONNREFUSED 127.0.0.1:3000\n    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)\nEmitted 'error' event on ClientRequest instance at:\n    at Socket.socketErrorListener (node:_http_client:500:9)\n    at Socket.emit (node:events:519:28)\n    at emitErrorNT (node:internal/streams/destroy:169:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:128:3)\n    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {\n  errno: -111,\n  code: 'ECONNREFUSED',\n  syscall: 'connect',\n  address: '127.0.0.1',\n  port: 3000\n}\n\nNode.js v20.16.0\n"
```

**Comment by TBG-FR** [comment]
> Hi all, We’re going to close this one due to inactivity. Please feel free to reopen if you’re still having issues here or open a new issue.

Issue is still there, and the PR/fix is #23300 😉
